_posts/2023-10-11-14.0.0-release.md

```diff
@@ -0,0 +1,132 @@
+
+In Go, the underlying generated Protobuf code is now exposed for easier low-level integrations with Flight ([GH-36893](https://github.com/apache/arrow/issues/36893)). 
+
+In Java, the stateful "login" authentication APIs using the Handshake RPC are deprecated; it will not be removed, but it should not be used unless you specifically want the old behavior ([GH-37722](https://github.com/apache/arrow/issues/37722)). Utilities were added to help implement basic Flight SQL services for unit testing ([GH-37795](https://github.com/apache/arrow/issues/37795)).
+
+## C++ notes
+
+## C# notes
+
+## Go notes
+
+## Java notes
+
+Java 21 is enabled and validated in CI ([GH-37914](https://github.com/apache/arrow/issues/37914)).
+
+The Gandiva module implemented a breaking change by moving `Types.proto` into a subfolder ([GH-37893](https://github.com/apache/arrow/issues/37893)).
+
+`DefaultVectorComparators` added support for `LargeVarCharVector`, `LargeVarBinaryVector` ([GH-25659](https://github.com/apache/arrow/issues/25659)) and for `BitVector`, `DateDayVector`, `DateMilliVector`
+`Decimal256Vector`, `DecimalVector`, `DurationVector`, `IntervalDayVector`, `TimeMicroVector`, `TimeMilliVector`, `TimeNanoVector`, `TimeSecVector`, `TimeStampVector` ([GH-37701](https://github.com/apache/arrow/issues/37701)).
+
+A bug was fixed in `VectorAppender` to prevent resizing the data buffer twice when appending variable-length vectors ([GH-37829](https://github.com/apache/arrow/issues/37829)).
+
+`VarCharWriter` added support for writing from `Text` and `String` ([GH-37706](https://github.com/apache/arrow/issues/37706)). `VarBinaryWriter` added support for writing from `byte[]` and `ByteBuffer` ([GH-37705](https://github.com/apache/arrow/issues/37705)).
+
+The JDBC driver will now ignore username and password authentication if a token is provided ([GH-37073](https://github.com/apache/arrow/issues/37073)).
+
+A bug was fixed in the Java C-Data interface when importing a vector with an empty array ([GH-37056](https://github.com/apache/arrow/issues/37056)).
+
+A bug was fixed in the S3 file system implementation when closing the connection ([GH-36069](https://github.com/apache/arrow/issues/36069)).
+
+Arrow datasets now support Substrait `ExtendedExpression`s as inputs to filter and project operations ([GH-34252](https://github.com/apache/arrow/issues/34252)).
+
+## JavaScript notes
+
+* GH-21815: [JS] Add support for Duration type #37341
+* GH-31621: [JS] Fix Union null bitmaps #37122
+
+## Python notes
```

Comment:
```suggestion
## Python notes

Compatibility notes:
* Support for Python 3.12 was added [GH-37880](https://github.com/apache/arrow/issues/37880)
* Support for Cython 3 was added [GH-37742](https://github.com/apache/arrow/issues/37742)
* PyArrow is now compatible with numpy 2.0 [GH-37574](https://github.com/apache/arrow/issues/37574)
* `pyarrow.compute.CumulativeSumOptions` has been deprecated, use `pyarrow.compute.CumulativeOptions` instead [GH-36240](https://github.com/apache/arrow/issues/36240)

New features:
* Allow type promotion added on `pyarrow.concat_tables` [GH-36845](https://github.com/apache/arrow/issues/36845)
* Support for vector function UDF was added [GH-36672](https://github.com/apache/arrow/issues/36672)

Other improvements:
* `pyarrow.MapScalar.as_py`can now be called with custom field name [GH-36809](https://github.com/apache/arrow/issues/36809)
* The default of `pre_buffer` is now set to `True` for reading Parquet when using `pyarrow.dataset` directly. This can give significant speed-up on filesystems like S3 and is now aligned to `pyarrow.parquet.read_table` interface [GH-36765](https://github.com/apache/arrow/issues/36765)
* Path to timezone database can now be set through python API ([GH-35600](https://github.com/apache/arrow/issues/35600), [GH-38145] (https://github.com/apache/arrow/issues/38145))

Relevant bug fixes:
* String to date cast kernel was added to fix python scalar cast regression [GH-37411](https://github.com/apache/arrow/issues/37411)
* Fix conversion from Python to Arrow when chunking large nested structs [GH-32439](https://github.com/apache/arrow/issues/32439)
* Fix segfault when passing table as argument to `pyarrow.Table.filter` [GH-37650](https://github.com/apache/arrow/issues/37650)
* `use_threads` keyword was added to the `group_by` method on `pyarrow.Table` which gets passed through to the `pyarrow.acero.Declaration.to_table` call. Specifing `use_threads=False`allows to get stable ordering of the output [GH-36709](https://github.com/apache/arrow/issues/36709)
* Fix printable representation for `pyarrow.TimestampScalar` when values are outside datetime range [GH-36323](https://github.com/apache/arrow/issues/36323)
* Empty dataframes with zero chunks can now be consumed by the Dataframe Interchange Protocol implementation [GH-37050](https://github.com/apache/arrow/issues/37050) 
* Fix dtype information for categorical columns in the Dataframe Interchange Protocol implementation [GH-38034](https://github.com/apache/arrow/issues/38034)
* Boolean columns with bitsize 1 are now supported in `from_dataframe`of the Dataframe Interchange Protocol [GH-37145](https://github.com/apache/arrow/issues/37145)

```